sound/soc/sof/sof-client.c

```diff
@@ -284,6 +285,20 @@ int sof_client_ipc_set_get_data(struct sof_client_dev *cdev, void *ipc_msg,
 }
 EXPORT_SYMBOL_NS_GPL(sof_client_ipc_set_get_data, SND_SOC_SOF_CLIENT);
 
+struct sof_ipc4_fw_module *sof_client_ipc4_find_module_by_uuid(struct sof_client_dev *cdev, const guid_t *uuid)
+{
+	struct snd_sof_dev *sdev = cdev->sdev;
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_IPC4)
+	if (sdev->pdata->ipc_type == SOF_INTEL_IPC4)
+		return sof_ipc4_find_module_by_uuid(sdev, uuid);
+#endif
+	dev_err(sdev->dev, "%s: IPC4 required", __func__);
```

Comment:
I think whether IPC4 support is built and whether its in use are two different things. The ifdef is there to avoid undefined symbol if IPC4 support is not built. If statement maybe unnecessary if sof_ipc4_find_module_by_uuid() behaves well even if IPC4 is not in use, but to me it does not look like that. Will update the error print.